src/DataTable/SmartStatus.jsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { requiredWhen } from './utils/propTypesUtils';
+import SelectionStatus from './SelectionStatus';
+import RowStatus from './RowStatus';
+import FilterStatus from './FilterStatus';
+
+const SMART_STATUS_CLASS = 'pgn__smart-status';
+
+const SmartStatus = ({
+  isSelectable,
+  numberOfSelectedRows,
+  toggleAllRowsSelected,
+  isFilterable,
+  filterNames,
+  resetAllFilters,
+  pageSize,
+  itemCount,
+}) => {
+  if (isSelectable && numberOfSelectedRows > 0) {
+    return (
+      <SelectionStatus
+        numberOfSelectedRows={numberOfSelectedRows}
+        toggleAllRowsSelected={toggleAllRowsSelected}
+        itemCount={itemCount}
+        className={SMART_STATUS_CLASS}
+      />
+    );
+  }
+  if (isFilterable && filterNames.length > 0) {
+    return (
+      <FilterStatus
+        className={SMART_STATUS_CLASS}
+        filterNames={filterNames}
+        onClick={() => resetAllFilters()}
+      />
+    );
+  }
+  return <RowStatus className={SMART_STATUS_CLASS} pageSize={pageSize} itemCount={itemCount} />;
+};
+
+SmartStatus.defaultProps = {
+  numberOfSelectedRows: 0,
+  toggleAllRowsSelected: () => {},
+  filterNames: [],
+};
+
+SmartStatus.propTypes = {
+  isSelectable: PropTypes.bool.isRequired,
+  numberOfSelectedRows: requiredWhen(PropTypes.number, 'isSelectable'),
```

Comment:
Why go through the trouble of `requiredWhen` since we have the reasonable default of 0?

Reply:
Generally speaking I think if we find we need requiredWhen a lot we should consider if there's an alternative way to organize or nest our components

Reply:
Yeah, I'm generally using it when react-table has requirements that are dependent on others. But you're right, here there's a very reasonable default.